src/cargo/ops/cargo_compile/unit_generator.rs

```diff
@@ -267,35 +268,61 @@ impl<'a> UnitGenerator<'a, '_> {
                 })
                 .collect::<Vec<_>>();
             let suggestion = closest_msg(target_name, targets.iter(), |t| t.name(), "target");
-            if !suggestion.is_empty() {
-                anyhow::bail!(
-                    "no {} target {} `{}`{}",
-                    target_desc,
-                    if is_glob { "matches pattern" } else { "named" },
-                    target_name,
-                    suggestion
-                );
+            let target_elsewhere = if is_glob {
+                let pattern = build_glob(target_name)?;
+                let filter = |t: &Target| is_expected_kind(t) && pattern.matches(t.name());
+                self.get_targets_from_other_packages(filter)?
             } else {
-                let mut msg = String::new();
-                writeln!(
-                    msg,
-                    "no {} target {} `{}`.",
-                    target_desc,
-                    if is_glob { "matches pattern" } else { "named" },
-                    target_name,
-                )?;
-                if !targets.is_empty() {
-                    writeln!(msg, "Available {} targets:", target_desc)?;
-                    for target in targets {
-                        writeln!(msg, "    {}", target.name())?;
-                    }
+                let filter = |t: &Target| t.name() == target_name && is_expected_kind(t);
+                self.get_targets_from_other_packages(filter)?
+            };
+
+            let mut msg = String::new();
+            write!(
+                msg,
+                "no {} target {} `{}`",
+                target_desc,
+                if is_glob { "matches pattern" } else { "named" },
+                target_name
+            )?;
+            if let Some((package, target)) = target_elsewhere {
+                writeln!(msg, "\nAvailable {target_desc} targets in {package}:")?;
```

Comment:
Could we make `in {package}` its own commit as well?  That is an independent improvement